src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -979,11 +988,10 @@ def enable_model_cpu_offload(self, gpu_id: Optional[int] = None, device: Union[t
         device = torch.device(f"{device_type}:{self._offload_gpu_id}")
         self._offload_device = device
 
-        if self.device.type != "cpu":
```

Comment:
`self.device.type == "cpu"` does not mean all the models are on cpu device, no?

Reply:
Yes. This checks if the pipeline has been moved to GPU before `enable_model_cpu_offload` has been called. If this is the case, it moves the whole pipeline to CPU first. But you're right that it's not needed since all components are moved to CPU anyway.